runtime/src/common/quantity.rs

```diff
@@ -67,6 +67,117 @@ impl<'a> AddAssign<&'a Quantity> for Quantity {
     }
 }
 
+impl AddAssign<Quantity> for Quantity {
+    fn add_assign(&mut self, other: Quantity) {
+        self.0 += other.0;
+    }
+}
+
+impl Add<u64> for Quantity {
+    type Output = Quantity;
+
+    fn add(mut self, other: u64) -> Quantity {
+        self += other;
+        self
+    }
+}
+
+impl AddAssign<u64> for Quantity {
+    fn add_assign(&mut self, other: u64) {
+        self.0 += other;
+    }
+}
+
+impl Mul for Quantity {
+    type Output = Quantity;
+
+    fn mul(mut self, rhs: Quantity) -> Quantity {
+        self *= &rhs;
+        self
+    }
+}
+
+impl<'a> Mul<&'a Quantity> for Quantity {
+    type Output = Quantity;
+
+    fn mul(mut self, rhs: &Quantity) -> Quantity {
+        self *= rhs;
+        self
+    }
+}
+
+impl<'a> MulAssign<&'a Quantity> for Quantity {
+    fn mul_assign(&mut self, rhs: &Quantity) {
+        self.0 *= &rhs.0;
+    }
+}
+
+impl MulAssign<Quantity> for Quantity {
+    fn mul_assign(&mut self, rhs: Quantity) {
+        self.0 *= rhs.0;
+    }
+}
+
+impl Mul<u64> for Quantity {
+    type Output = Quantity;
+
+    fn mul(mut self, other: u64) -> Quantity {
+        self *= other;
+        self
+    }
+}
+
+impl MulAssign<u64> for Quantity {
+    fn mul_assign(&mut self, other: u64) {
+        self.0 *= other;
+    }
+}
+
+impl Div for Quantity {
```

Comment:
Should we consider instead implementing `num_traits::ops::checked::CheckedDiv` similar as with Sub?

Reply:
Good idea, changed.

Reply:
cool! looks good :+1: